File: frontend/src/types/property.ts
export interface Property {
  id: number;
  source: string;
  external_id?: string;

  title: string;
  project_name?: string;
  developer?: string;

  listing_type?: string;
  property_type?: string;

  country?: string;
  city?: string;
  location?: string;

  price?: number;
  currency?: string;

  bedrooms?: number;
  bathrooms?: number;

  area?: number;
  area_unit?: string;

  status?: string;
  completion_date?: string;

  description?: string;

  amenities?: string[];
  brands?: string[];
  images?: string[];

  source_url: string;
}

export interface PropertyMetadata {
  property_name?: string;
  developer?: string;
  brand?: string;
  country?: string;
  city?: string;
  location?: string;
  property_type?: string;
  investment_type?: string;
  description?: string;
  source_url?: string;
  price?: string;
  bedrooms?: string;
  area?: string;
  title?: string;
}

export interface PropertyIntelligence {
  summary: string;

  strengths: string[];

  risks: string[];

  infrastructure: {
    score: number;
    assessment: string;
    strengths: string[];
    weaknesses: string[];
  };

  pricing: {
    available: boolean;
    comparable_count: number;
    priced_comparables: number;
    average_price?: number;
    median_price?: number;
    average_price_per_area?: number;
  };

  data_completeness: number;

  confidence: number;
}

export interface PropertyAnalysis {
  property_id: number;
  intelligence: PropertyIntelligence;
}

export interface PropertyResult {
  id?: number;
  document: string;
  metadata: PropertyMetadata;
}

File: backend/app/schemas/property.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class PropertyBase(BaseModel):
    source: str
    external_id: str | None = None

    title: str
    project_name: str | None = None
    developer: str | None = None

    listing_type: str | None = None
    property_type: str | None = None

    country: str | None = None
    city: str | None = None
    location: str | None = None

    price: float | None = None
    currency: str | None = None

    bedrooms: int | None = None
    bathrooms: int | None = None

    area: float | None = None
    area_unit: str | None = None

    status: str | None = None
    completion_date: str | None = None

    description: str | None = None

    amenities: list[str] | None = None
    brands: list[str] | None = None
    images: list[str] | None = None

    source_url: str
    raw_content: str | None = None

    latitude: float | None = None
    longitude: float | None = None



class PropertyResponse(PropertyBase):
    id: int
    scraped_at: datetime

    model_config = ConfigDict(from_attributes=True)

class ChatRequest(BaseModel):
    message: str


class ChatResponse(BaseModel):
    message: str
    intent: str
    filters: dict
    results: list[dict]
    semantic_results: list[dict]
    analysis: list[dict]
    investment: list[dict] = []